Formula/cp2k.rb

```diff
@@ -35,47 +38,95 @@ def install
-    ENV["LIBINT_INCLUDE_DIR"] = libexec/"include"
-    ENV["LIBINT_LIB_DIR"] = libexec/"lib"
-
-    # CP2K configuration is done through editing of arch files
-    inreplace Dir["arch/Darwin-gfortran.*"].each do |s|
-      s.gsub!(/DFLAGS *=/, "DFLAGS = -D__FFTW3")
-      s.gsub!(/FCFLAGS *=/, "FCFLAGS = -I#{Formula["fftw"].opt_include}")
-      s.gsub!(/LIBS *=/, "LIBS = #{libs.join(" ")}")
-    end
+    arch = "local"
+    if OS.mac?
+      arch = "Darwin-gfortran"
+
+      # libint needs `-lstdc++` (https://github.com/cp2k/cp2k/blob/master/INSTALL.md)
+      # Can remove if added upstream to Darwin-gfortran.psmp and Darwin-gfortran.ssmp
+      libs = %W[
+        -L#{Formula["fftw"].opt_lib}
+        -lfftw3
+        -lstdc++
+      ]
+
+      ENV["LIBXC_INCLUDE_DIR"] = Formula["libxc"].opt_include
+      ENV["LIBXC_LIB_DIR"] = Formula["libxc"].opt_lib
+      ENV["LIBINT_INCLUDE_DIR"] = libexec/"include"
+      ENV["LIBINT_LIB_DIR"] = libexec/"lib"
 
-    # MPI versions link to scalapack
-    inreplace Dir["arch/Darwin-gfortran.p*"],
-              /LIBS *=/, "LIBS = -L#{Formula["scalapack"].opt_prefix}/lib"
+      # CP2K configuration is done through editing of arch files
+      inreplace Dir["arch/Darwin-gfortran.*"].each do |s|
+        s.gsub!(/DFLAGS *=/, "DFLAGS = -D__FFTW3")
+        s.gsub!(/FCFLAGS *=/, "FCFLAGS = -I#{Formula["fftw"].opt_include}")
+        s.gsub!(/LIBS *=/, "LIBS = #{libs.join(" ")}")
+      end
 
-    # OpenMP versions link to specific fftw3 library
-    inreplace Dir["arch/Darwin-gfortran.*smp"],
-              "-lfftw3", "-lfftw3 -lfftw3_threads"
+      # MPI versions link to scalapack
+      inreplace Dir["arch/Darwin-gfortran.p*"],
+                /LIBS *=/, "LIBS = -L#{Formula["scalapack"].opt_lib}"
+
+      # OpenMP versions link to specific fftw3 library
+      inreplace Dir["arch/Darwin-gfortran.*smp"],
+                "-lfftw3", "-lfftw3 -lfftw3_threads"
+    else
+      args = %W[
+        -j #{ENV.make_jobs}
+        --mpi-mode=openmpi
+        --math-mode=openblas
+        --with-gcc=system
+        --with-intel=no
+        --with-cmake=no
+        --with-openmpi=#{Formula["open-mpi"].opt_prefix}
+        --with-mpich=no
+        --with-intelmpi=no
+        --with-libxc=#{Formula["libxc"].opt_prefix}
+        --with-libint=#{libexec}
+        --with-fftw=#{Formula["fftw"].opt_prefix}
+        --with-acml=no
+        --with-mkl=no
+        --with-openblas=#{Formula["openblas"].opt_prefix}
+        --with-scalapack=#{Formula["scalapack"].opt_prefix}
+        --with-libxsmm=no
+        --with-elpa=no
+        --with-ptscotch=no
+        --with-superlu=no
+        --with-pexsi=no
+        --with-quip=no
+        --with-plumed=no
+        --with-sirius=no
+        --with-gsl=no
+        --with-libvdwxc=no
+        --with-spglib=no
+        --with-hdf5=no
+        --with-spfft=no
+        --with-spla=no
+        --with-cosma=no
+        --with-libvori=no
+      ]
+      args << "--generic" if build.bottle?
```

Comment:
Probably not necessary. Default is `-mtune=native` but superenv probably drops it. Same thing for `--generic` arg which sets `-mtune=generic`.
```suggestion
```